velox/type/tests/DecimalTest.cpp

```diff
@@ -176,5 +176,96 @@ TEST(DecimalTest, toByteArray) {
   testToByteArray(DecimalUtil::kLongDecimalMax, expected8, 16);
 }
 
+template <typename T>
+void checkRescaleDouble(double value, const TypePtr& type, T expectedValue) {
+  std::string error;
+  auto [precision, scale] = getDecimalPrecisionScale(*type);
+  std::optional<T> result =
+      DecimalUtil::rescaleDouble<T>(value, precision, scale, error);
+  ASSERT_TRUE(result.has_value());
+  ASSERT_TRUE(error.empty());
+  ASSERT_EQ(result.value(), expectedValue);
+}
+
+template <typename T>
+void checkRescaleDoubleFails(
+    double value,
+    const TypePtr& type,
+    const std::string& expectedError) {
+  std::string error;
+  auto [precision, scale] = getDecimalPrecisionScale(*type);
+  std::optional<T> result =
+      DecimalUtil::rescaleDouble<T>(value, precision, scale, error);
+  ASSERT_FALSE(result.has_value());
+  ASSERT_FALSE(error.empty());
+  ASSERT_EQ(error, expectedError);
+}
+
+TEST(DecimalTest, rescaleDouble) {
+  checkRescaleDouble<int64_t>(-3333.03, DECIMAL(10, 4), -33'330'300);
+  checkRescaleDouble<int128_t>(-3333.03, DECIMAL(20, 1), -33'330);
+  checkRescaleDouble<int128_t>(-3333.03, DECIMAL(20, 10), -33'330'300'000'000);
+
+  checkRescaleDouble<int64_t>(-2222.02, DECIMAL(10, 4), -22'220'200);
+  checkRescaleDouble<int128_t>(-2222.02, DECIMAL(20, 1), -22'220);
+  checkRescaleDouble<int128_t>(-2222.02, DECIMAL(20, 10), -22'220'200'000'000);
+
+  checkRescaleDouble<int64_t>(-1.0, DECIMAL(10, 4), -10'000);
+  checkRescaleDouble<int128_t>(-1.0, DECIMAL(20, 1), -10);
+  checkRescaleDouble<int128_t>(-1.0, DECIMAL(20, 10), -10'000'000'000);
+
+  checkRescaleDouble<int64_t>(0.00, DECIMAL(10, 4), 0);
+  checkRescaleDouble<int128_t>(0.00, DECIMAL(20, 1), 0);
+  checkRescaleDouble<int128_t>(0.00, DECIMAL(20, 10), 0);
+
+  checkRescaleDouble<int64_t>(100, DECIMAL(10, 4), 1'000'000);
+  checkRescaleDouble<int128_t>(100, DECIMAL(20, 1), 1'000);
+  checkRescaleDouble<int128_t>(100, DECIMAL(20, 10), 1'000'000'000'000);
+
+  checkRescaleDouble<int64_t>(99999.99, DECIMAL(10, 4), 999'999'900);
+  checkRescaleDouble<int128_t>(99999.99, DECIMAL(20, 1), 1'000'000);
+  checkRescaleDouble<int128_t>(99999.99, DECIMAL(20, 10), 999'999'900'000'000);
+
+  checkRescaleDouble(std::numeric_limits<double>::min(), DECIMAL(38, 2), 0);
+
+  const std::string messageForInfiniteValue = "Value is not finite.";
```

Comment:
- messageFor is redundant and makes the variable name unnecessarily long
- constants should be defined as static and have names starting with 'k'

```
static const std::string kValueNotFinite = "...";
static const std::string kOverflow = "...";
```